### PR TITLE
Removed `create_function()` function usage

### PR DESCRIPTION
**Pull request by YevSent**
Removed `create_function()` function usage

### Fixed Issues
1. magento-engcom/php-7.2-support#8: Deprecated: Function create_function() is deprecated

Changes for EE in the internal branch https://github.com/magento-mpi/magento2ee/tree/G%238

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: lib/internal/Magento/Framework/Validator/Test/Unit/Constraint/Option/CallbackTest.php [hunk 1]
```diff
@@ -28,41 +27,46 @@ class CallbackTest extends \PHPUnit\Framework\TestCase
...
-        )->with(
-            'arg1',
-            'arg2'
-        )->will(
-            $this->returnValue('Value from mock')
-        );
+        $mock = $this->getMockBuilder('Foo')
+            ->setMethods(['getValue'])
+            ->getMock();
+        $mock->method('getValue')
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/Validator/Test/Unit/Constraint/Option/CallbackTest.php:46]: Looks like section `->expects($this->once())` is missing here

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/Validator/Test/Unit/Constraint/Option/CallbackTest.php (hunk 1 above)
Comment (YevSent) [reply lib/internal/Magento/Framework/Validator/Test/Unit/Constraint/Option/CallbackTest.php:46]: It doesn't matter how many times `getValue` will be called, that's why `expects` is not needed.

Comment (ihor-sviziev) [reply lib/internal/Magento/Framework/Validator/Test/Unit/Constraint/Option/CallbackTest.php:46]: In this case would be better to use `->expects($this->any())` to make it clear

Comment (YevSent) [reply lib/internal/Magento/Framework/Validator/Test/Unit/Constraint/Option/CallbackTest.php:46]: For this test case, in general, it doesn't matter how many times the method will be called, so `any()` is redundant. The `expects()` may sense to use when we need to know if the method is called or not (corner cases, like business logic conditionals) or how many times it was called - for performance reasons.
I don't see, how `any()` makes test a cleaner.

Comment (ihor-sviziev) [reply lib/internal/Magento/Framework/Validator/Test/Unit/Constraint/Option/CallbackTest.php:46]: Ok. Good point. Let's leave it as is

**Review by ihor-sviziev — APPROVED** [review]

**Review by buskamuza — APPROVED** [review]
